### PR TITLE
Order index by total_score

### PR DESCRIPTION
**Pull request by notbakaneko**
Orders the index by `total_score` which according to Elastic, can improve the performance of querying aggregations with a sort order since lucene can assume when it no longer has to load extra segments.

This needs actual testing with a reasonably sized dataset to actually compare with the unordered index.

Also changes some of the fields to `keyword` as those are supposedly better for term filtering when ranges are not used.

### CODE REVIEW

**Comment by peppy** [comment]
@nanaya can you run this against your local dataset while we're still waiting on the staging one and see if you can notice a difference?

File: osu.ElasticIndexer/schemas/solo_scores.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
         "type": "date"
       },
       "id": {
-        "type": "long"
+        "type": "keyword"
```
Comment (nanaya) [inline osu.ElasticIndexer/schemas/solo_scores.json:29]: I think the id should still be long because it's used for the tie breaker.

**Review by nanaya — COMMENTED** [review]
I wonder if the sort settings will have any effect on user best scores section as that one is sorted by pp instead of total scores

**Comment by peppy** [comment]
> I wonder if the sort settings will have any effect on user best scores section as that one is sorted by pp instead of total scores

In theory it would be *more in order* with this change, but it's definitely an interesting question. Going forward we'll also be showing beatmap leaderboards using `pp` order, so this may also be more relevant than just user profiles.

**Comment by smoogipoo** [comment]
As an aside question: in the future we're going to have to convert all scores to lazer's "standardised scoring", and the position of scores will change a little bit compared to if ordering via `total_score`.
We'll still want to retain `total_score`, maybe by renaming it to a `legacy_total_score`, but the question I have is: will this pose any problems?

**Comment by peppy** [comment]
Shouldn't do. We'd run a reindex at that point and it would sort itself out.

File: osu.ElasticIndexer/schemas/solo_scores.json (hunk 1 above)
Comment (notbakaneko) [reply osu.ElasticIndexer/schemas/solo_scores.json:29]: `keyword` works fine for sort and cursor

Comment (notbakaneko) [reply osu.ElasticIndexer/schemas/solo_scores.json:29]: of course this makes the assumption that no one tries to enter a string into the field and mess it up 🤔

**Review by nanaya — APPROVED** [review]

**Pull request merged**